src/collection.rs

```diff
@@ -442,29 +454,208 @@ impl<'a, C: ClientExt> Collection<'a, C> {
+    /// operation to disk even in case that the waitForSync flag had been disabled for the entire collection.
+    /// Thus, the waitForSync query parameter can be used to force synchronization of just this specific operations.
+    /// To use this, set the waitForSync parameter to true. If the waitForSync parameter is not specified or set to false,
+    /// then the collection’s default waitForSync behavior is applied.
+    /// The waitForSync query parameter cannot be used to disable synchronization for collections that have a default waitForSync value of true.
+    ///
+    /// If the query parameter returnNew is true, then, for each generated document, the complete new document is returned under the new attribute in the result.
     /// # Note
     /// this function would make a request to arango server.
     #[maybe_async]
-    pub async fn create_document<T>(&self, _doc: Document<T>) {
-        unimplemented!()
+    pub async fn create_document<T>(
+        &self,
+        doc: T,
+        insert_options: Option<DocumentInsertOptions>,
+    ) -> Result<DocumentResponse<T>, ClientError>
+    where
+        T: Serialize + DeserializeOwned,
+    {
+        let mut url = self.document_base_url.join("").unwrap();
+        let body = serde_json::to_string(&doc)?;
+
+        if let Some(options) = insert_options {
+            if let Some(return_new) = options.borrow().return_new {
+                url.query_pairs_mut()
+                    .append_pair("returnNew", return_new.to_string().as_str());
+            }
+            if let Some(wait_for_sync) = options.borrow().wait_for_sync {
+                url.query_pairs_mut()
+                    .append_pair("waitForSync", wait_for_sync.to_string().as_str());
+            }
+            if let Some(return_old) = options.borrow().return_old {
+                url.query_pairs_mut()
+                    .append_pair("returnOld", return_old.to_string().as_str());
+            }
+            if let Some(silent) = options.borrow().silent {
+                url.query_pairs_mut()
+                    .append_pair("silent", silent.to_string().as_str());
+            }
+            if let Some(overwrite) = options.borrow().overwrite {
+                url.query_pairs_mut()
+                    .append_pair("overwrite", overwrite.to_string().as_str());
+            }
+            #[cfg(feature = "arango3_7")]
+            if let Some(overwrite_mode) = options.overwrite_mode {
+                let mode = match overwrite_mode {
+                    DocumentOverwriteMode::Ignore => "ignore",
+                    DocumentOverwriteMode::Replace => "replace",
+                    DocumentOverwriteMode::Update => "update",
+                    DocumentOverwriteMode::Conflict => "conflict",
+                };
+                url.query_pairs_mut().append_pair("overwriteMode", mode);
+            }
+        }
+
+        let resp: DocumentResponse<T> =
```

Comment:
I have updated the dev branch. The last commit should help correctly deserialize error response related to document. With this commit, use `deserialize_response` here and anywhere that deserilize response from server.